src/CodeXMLParse.cpp

```diff
@@ -143,11 +146,39 @@ void AnnotateColor(ANNOTATOR_PARAMS)
 	out->push_back(annotation);
 }
 
+void AnnotateVariable(ANNOTATOR_PARAMS)
```

Comment:
I thought we didn't want to handle string literals.

Reply:
String literals are not in the address space "ram".

Reply:
Take a look at the address. It is a correct address of string literal.

Reply:
From our discussion regarding this, my understanding was to take constant variables and global variables that are in the address space "ram".

Reply:
Here's what I could find about string literals so far

**Case 1**: String literals with no modification to related parts

When we have a string literal without any changes to the data(flags) at its address, that string literal is shown. Ghidra creates a varnode for it, and from the varnode we can find that the type of the address is 'pointer'. Hence, in this case, we can give the offset given by this varnode to the annotation for the constant string literal. This can be used for Xrefs and for the Show-in action. 
XML line is 
Line 490: `<variable color="const" varref="0x75f" opref="0x1db">"pure"</variable>`

**Case 2: When the flag of the string literal is deleted.**

The location of the string literal will be shown. For example, sym.imp.printf("pure") becomes sym.imp.printf(0x804a01b). But the varnode of the "0x804a01b" in this case points to the literal constant `0x804a01b`. This means it doesn't give any idea as to if it's a pointer or not.

XML line is 
Line 475: `<variable color="const" varref="0x741" opref="0x103">0x804a01b</variable>`

****Case 3**: When a new flag is added after deleting existing flag**

`sym.imp.printf(0x804a01b)` becomes `sym.imp.printf(<name of flag>)`. 

XML line
`<variable color="global" opref="0x1db">NameOfFlag</variable>`
In this case, we don't have any reference to varref from here. Even though varref is not available in the line output line we get, we could go to the corresponding "PcodeOp *" to find references to three varnodes: (in case 3)

```
<op code="66">
<seqnum space="ram" offset="0x8049269" uniq="0x1db"/>
    <addr ref="0x74d"/>
    <addr ref="0x74c"/>
    <addr ref="0x74e"/>
</op>
```
Here, the third address reference refers to the string literal. However, reliably finding that the third reference is exactly what we are looking seems impossible to me. In this case, the first varnode also has metatype "pointer". 

In case 2, however, we don't have a reference to the correct varnode even in the PcodeOp. Here, the references we found in case 3, are from PcodeOp.output and PcodeOp.inrefs

...

Binary used in the new test is used for this analysis and the name of the function is `sym.PrintAmbassador`.

When I said type, I meant the metatype we can see in this part of the XML. (`varnode.high.type.metatype == TYPE_PTR`)
```
        <high repref="0x75f" class="constant">
          <type name="" size="4" metatype="ptr" core="true">
            <typeref name="char" id="0x8000fed063689fd9"/>
          </type>
          <addr ref="0x75f"/>
        </high>
```
XMLs are attached to this comment.

Reply:
Inspecting further, this what I think is a possible solution for recognizing (some) constant addresses. The types of variables are defined in `type.hh`. Here's the related part.
```
/// The core meta-types supported by the decompiler. These are sizeless templates
/// for the elements making up the type algebra. Ordering is important: The lower
/// the number, the more \b specific the type, in calculations involving the generality
/// of a type.
enum type_metatype {
  TYPE_VOID = 10,		///< Standard "void" type, absence of type
  TYPE_SPACEBASE = 9,		///< Placeholder for symbol/type look-up calculations
  TYPE_UNKNOWN = 8,		///< An unknown low-level type. Treated as an unsigned integer.
  TYPE_INT = 7,			///< Signed integer. Signed is considered less specific than unsigned in C
  TYPE_UINT = 6,		///< Unsigned integer
  TYPE_BOOL = 5,		///< Boolean
  TYPE_CODE = 4,		///< Data is actual executable code
  TYPE_FLOAT = 3,		///< Floating-point

  TYPE_PTR = 2,			///< Pointer data-type
  TYPE_ARRAY = 1,		///< Array data-type, made up of a sequence of "element" datatype
  TYPE_STRUCT = 0		///< Structure data-type, made up of component datatypes
};
```
Here, we could annotate variables that are constant and belong to the type `TYPE_PTR` as they will be having a valid address(I believe so, it's definitely true in the case of string literals). As of now, as I said above, we don't have any examples of constant variable annotations with the current code in this PR. 

If we annotate constant variables of the type `TYPE_PTR`, we can successfully have actions  `Show in at address` and `Xrefs` for such constants variables (that will include all string literals). 
@karliss I want your opinion on this.

Reply:
Sounds great. I am happy to seethat you were able to figure it out.

Reply:
Implemented this.